packages/amplify_api/lib/graphql_helpers.dart

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_api_plugin_interface/amplify_api_plugin_interface.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:flutter/foundation.dart';
+import './amplify_api.dart';
+
+class ModelQueries extends ModelQueriesInterface {
+  @override
+  static GraphQLRequest<T> get<T extends Model>(
+      ModelType modelType, String id) {
+    ModelSchema schema = GraphQLRequestFactory.getSchema(modelType);
+
+    String modelName = schema.name;
+    Map<String, ModelField?>? fieldsMap = schema.fields;
+
+    List<String> fieldsList = [];
```

Comment:
suggestion: I think this might be  place you can break this out into a smaller utility method, something like `List<String> getFieldsFromModelType(ModelType modelType) {}`. This will accomplish a few things:
* make this method easier to read bc this whole block will be reduced to `final fieldsList = getFieldsFromModelType...`, pretty clear what it does
* easier to use this method somewhere else, like maybe a mutation/subscription helper
* can test the utility

Reply:
Great points. I've started to use this functional approach throughout the helper as it makes sense.